classes/library.rb

```diff
@@ -0,0 +1,65 @@
+require_relative 'author'
+require_relative 'book'
+require_relative 'order'
+require_relative 'reader'
+require_relative 'seeds'
+require 'faker'
+require 'yaml'
+
+# Library stores all books, readers, authors and orders
+class Library
+  include Seeds
+
+  attr_accessor :books, :orders, :authors, :readers
+
+  def initialize(authors = [], books = [], orders = [], readers = [])
+    @books = books
+    @orders = orders
+    @readers = readers
+    @authors = authors
+  end
+
+  def write_to_yaml(file_name = "lib_#{self.object_id}.yml")
+    File.new(file_name, 'w') unless File.exist?(file_name)
+    File.open(file_name, 'w') { |file| file.write(self.to_yaml) }
+  end
+
+  def read_from_yaml(file_name = "lib_#{self.object_id}.yml")
+    if File.exist?(file_name)
+      library = YAML.load_file(file_name)
+      @authors = library.authors
+      @books = library.books
+      @orders = library.orders
+      @readers = library.readers
+    else
+      puts 'There is no such file.'
+    end
+  end
+
+  def most_popular_reader
+    most_popular(1, :reader).first.name
+  end
+
+  def count_readers_of_popular_books
+    most_popular(3, :book)
+    set = []
+    @orders.each { |order| set << order.reader if (order.book && books).any? }
```

Comment:
It is better to use select  orders with books and after then map to selects readers from selected orders. Then you don't need to use set = [], map returns array automatically